### PR TITLE
🏷️ Convert to typescript part 1

### PR DESCRIPTION
**Pull request by juliuxu**
Started converting a good chunk over to typescript.
I belive this will make it easier to make changes to the components in the future.

The changes should not break anything. The stories in storybook look fine, and I have used the components locally.

https://github.com/juliuxu/hedwig-remix-example/blob/main/app/routes/_index.tsx
image

In usage I get nice intellisense from the Typescript types ☺️ 
image

### CODE REVIEW

File: packages/accordion/src/AccordionItem.tsx [hunk 1]
```diff
@@ -51,7 +55,7 @@ const AccordionItem = ({ title, children, expanded }) => {
      * otherwise expandedState keeps track of which item is expanded
      */
     const showExpanded = () =>
-        expandedState.allowMultiple ? isExpanded : expandedState === title
+        expandedState.allowMultiple ? isExpanded : expandedState.open === title
```
Comment (juliuxu) [inline packages/accordion/src/AccordionItem.tsx:58]: 💡 This bug was caught by Typescript

File: .eslintrc [hunk 2]
```diff
@@ -1,9 +1,11 @@
 {
-    "parser": "babel-eslint",
+    "parser": "@typescript-eslint/parser",
```
Comment (juliuxu) [inline .eslintrc:2]: 🗒️ This lets eslint parse and lint the typescript files. Also underneath it adds some typescript specific eslint rules

File: packages/accordion/package.json [hunk 3]
```diff
@@ -30,10 +31,10 @@
         "url": "https://github.com/bring/hedwig-react-components/issues"
     },
     "dependencies": {
-        "@fortawesome/pro-solid-svg-icons": "^6.2.0",
+        "@fortawesome/fontawesome-svg-core": "^6.4.0",
```
Comment (juliuxu) [inline packages/accordion/package.json:34]: 🗒️ The package `@fortawesome/fontawesome-svg-core` has to be explicitly added to the dependencies
https://fontawesome.com/docs/web/use-with/react/#_1-add-svg-core

**Review by haraldvinje — APPROVED** [review]
Awesome 👏

**Review by viggo — APPROVED** [review]

**Review by viggo — APPROVED** [review]

**Pull request merged**